synapseclient/table.py

```diff
@@ -1756,6 +1756,15 @@ def from_data_frame(cls, schema, df, filepath=None, etag=None, quoteCharacter='"
 
             f = io.open(filepath, mode='w', encoding='utf-8', newline='')
 
+            import pandas as pd
```

Comment:
This sort of makes pandas a required install right?  Should there be a check for if pandas is installed?

Reply:
For consistency we should use the same check although it's in the from_data_frame which is likely only going to be called from a context where pandas was already used.

I'm not sure this is the best approach though, Godfrey was working on this as he left and didn't have a chance to finish it. At very least if we modify the df for the purposes of serializing it to a csv file we should copy it first. The to_csv function of the data frame has a date_format param on it to which we might be able to use instead although it I'm not sure there is an easy way to preserve millisecond accuracy using that string format alone...

Reply:
I'm late to this conversation, but what's the problem of requiring pandas?

Reply:
@mattfazza The python package actually doesn't require the installation of pandas.  So the parts of the client where pandas is needed `def asDataFrame(...)` actually uses this: https://github.com/Sage-Bionetworks/synapsePythonClient/blob/develop/synapseclient/table.py#L1455